src/app/pages/wishlist/wish-card/wish-card.component.html

```diff
@@ -16,7 +16,10 @@ <h1 class="wish-card__title">{{wishProduct.title | titlecase}}</h1>
                     <img src="assets/images/svg/details.svg" alt="Details">
                 </a>
             </button>
-            <button (click)="user ? handleAddToCart(wishProduct) : openDialog()">
+            <button *ngIf="user && objectKeys(user).length" (click)="handleAddToCart(wishProduct)">
```

Comment:
```suggestion
            <button *ngIf="user && objectKeys(user).length" (click)="handleAddToCart(wishProduct)">
```
As I told you before I think you can remove this check for keys in object (everywhere)